### ISSUE TITLE
rbac: unable to assign user permissions to add manage group members

### ISSUE THREAD

**Issue by JanKoppe** [goauthentik/authentik#8266]
**Describe your question/**
I'm trying to assign some user the permissions to manage the members of a specific group, by using the new RBAC features and assigning object permissions and global permissions

**Relevant infos**
Authentik 2023.10.6

**Screenshots**

Permissions of the user that should be able to manage the group members:

![image](https://github.com/goauthentik/authentik/assets/12087559/e4fed08b-3753-4bc5-8db8-59b4829cb413)

Screenshot of error message and failed (403) request when trying to add a new group member:

![image](https://github.com/goauthentik/authentik/assets/12087559/1b91657d-e430-4405-9082-2d5ea0d73843)

**Additional context**
From what I understand after reading the source code for a few minutes I'm think that I should be running into this GroupView action:

https://github.com/goauthentik/authentik/blob/main/authentik/core/api/groups.py#L149C1-L149C60

It's setup with required the `authentik_core.add_user` permission, which the user has assigned as a global permission. It also has the `authentik_core.change_group` permission on that specific Group object as far as I understand.

I'm not sure which permission is missing, or which permission check is failing here.

Also, I'm not sure if checking for `authentik_core.add_user` permission is correct here - shouldn't this rather be an additional permission to check (e.g. `authentic_core.add_group_member` and `authentic_core.remove_group_member`) similar to how additional permissions have been added on the User model? https://github.com/goauthentik/authentik/blob/main/authentik/core/models.py#L275

It seems like this method is requiren people to create new Users just to be able to add existing users to existing groups, which seems like unnecessarily wide permissions.

My use-case for this is that I want to grant some unprivileged users the permission to manage the members of specific groups. This would allow me to give owners of an application that uses Authentik for authentication to control who exactly can authenticate for that application, which we would like to use as a very binary authorization function.

Am I misunderstanding something here, or is this an actual issue with the very early preview of RBAC?

I really appreciate you adding the RBAC feature by the way, it would solve a huge issue and remove a bunch of work for us!

**Comment by esimes** [comment]
> Am I misunderstanding something here, or is this an actual issue with the very early preview of RBAC?

I just tested out this feature today aswell and expected that exact behaviour like you said. Basically thats how the most services like that work.  Glad I found out that im not alone. 

I also experimented with a few additional permissions, such as "Core Can Change User" but these did not work either. Therefore, I'm starting to think this might be a bug in the RBAC system.

From me also great feature. keep up the great work!

**Comment by JanKoppe** [comment]
@hiteshsondhi88 thanks for the suggestion. We've already saw that as well, but this is a complete no-go: This will allow those group managers to create groups with superadmin privileges and thus elevate their privileges. Security-wise, this would be a complete no-go.

**Comment by lectrician1** [comment]
@JanKoppe `Can add Group` doesn't allow users to create groups. If they try they just get a loading screen:

![image](https://github.com/goauthentik/authentik/assets/22477298/29979b90-ffde-4164-8abb-8a27018d62fd)

It's poorly named and really means "can add user to group"

**Comment by JanKoppe** [comment]
They can do it via the API, e.g. with the Terraform Provider. We've tried that, and it works without any issues.

The reason why the WebUI breaks is because it tries to fetch other information first to which the user does not have permissions. 

This will not prevent someone who wants to abuse this from doing so.

Example code for Terraform (written from memory, not tested, but that's the gist):
```
data "authentik_user" "me" {
  user = "maliciousperson"
}

resource "authentik_group" "hijack" {
  name = "allyourbasearebelongtome"
  users = [data.authentik_user.me.id]
  is_superuser = true
}
```

**Comment by esimes** [comment]
```
curl -v -X POST "https://sso.yourdomain.com/api/v3/core/groups/" \
 -H "accept: application/json" \
 -H "content-type: application/json" \
 -H "Authorization: Bearer apitokenofuser" \
 -d '{"name":"supermaliciousgroup","is_superuser":true,"attributes":{}}'
```
Here is a quick curl command to check that. Jan is right. The right "Can add group" gives via API the right to create superuser groups. 

After removing the permission again the user gets the error.
`{"detail":"You do not have permission to perform this action."} `

**Comment by BeryJu** [comment]
I forgot to link it, but this is fixed by #9254

**Comment by marcportabellaclotet-mt** [comment]
Nice feature!
Would it be possible to restrict the visibility of a subset of users through permissions? Okta recently introduced the concept of [realms](https://help.okta.com/oie/en-us/content/topics/users-groups-profiles/realms/realms.htm), enhancing user management security for large organizations.

**Comment by BeryJu** [comment]
You can already restrict the visibility of users by assigning permissions to specific users, however we're also planning to have an additional container object most likely called "Organizations" to further group users and manage them in chunks

**Comment by marcportabellaclotet-mt** [comment]
Organizations will help a lot when managing governance. Thanks

**Comment by Aetherinox** [comment]
I'm getting a similar issue, but the problem with mine is that it's just a blank error.

![5zgw5cu4j8](https://github.com/goauthentik/authentik/assets/118329232/def060cd-ec01-42be-9f7b-894dbf857efd)

Can't assign a user at all, just throws a blank message, and no other info in the logs.

**Comment by nate3D** [comment]
I'm having the same error as @Aetherinox . Just a blank error box in the client and a 400 back from the server. Not being able to assign roles is a big roadblock.

**Comment by Aetherinox** [comment]
God I wish I could figure out what came of this. I eventually got past it. Going to have to go mess with it again and see.
Originally, I believe it had something to do with the integrated outpost acting up, and I was attempting to add my user to a group and it was failing.

**Comment by nate3D** [comment]
I opened a new issue https://github.com/goauthentik/authentik/issues/10983 since this one is closed. I was looking at the PATCH request and I was able to get a better error message going through the API as the UI is eating the 400 response without displaying it. The issue seems to be "by design"? It won't allow a role to be assigned to more than one group at a time. That doesn't make any sense to me and is opposite of the documentation.
